src/Var.cc

```diff
@@ -195,18 +195,9 @@ static void make_var(const IDPtr& id, TypePtr t, InitClass c, ExprPtr init,
 		{
 		// This can happen because the grammar allows any "init_class",
 		// including none, to be followed by an expression.
-		// Remove in v6.1 (make an error)
-		reporter->Deprecation(
-			util::fmt("Remove in v6.1. Initialization not preceded by =/+=/-= is deprecated. (%s)",
-		              obj_desc_short(init.get()).c_str()),
-			init->GetLocationInfo());
-
-		// The historical instances of these, such as the
-		// language/redef-same-prefixtable-idx.zeek btest, treat
-		// this as += rather than =, and with the initializer
-		// implicitly inside a list.
-		init = make_intrusive<ListExpr>(init);
-		c = INIT_EXTRA;
+		reporter->Error(util::fmt("Initialization not preceded by =/+=/-= is not allowed. (%s)",
+		                          obj_desc_short(init.get()).c_str()),
+		                init->GetLocationInfo());
```

Comment:
Haven't tried the effect, but could we `init->Error()` or so and return early? Seems any following error wouldn't (yet) be relevant or maybe even confusing?

Or tighten the parser to cause a syntax error, but the descriptive message here is likely more useful..

Reply:
This is the output difference from `language.init-mismatch` after changing it to `init->Error()` and returning. Because of the return the `b` variable doesn't exist anymore and we get an extra error message. Without the return, the variable would have been created but just wouldn't have been initialized.

```
  == Diff ===============================
  --- /dev/fd/63	2023-06-13 19:06:37
  +++ /dev/fd/62	2023-06-13 19:06:37
  @@ -18,3 +18,4 @@
   error in <...>/init-mismatch.zeek, line 34: type clash in assignment (s3 = set(s2))
   error in <...>/init-mismatch.zeek, line 36: pattern += op requires op to be a pattern (p += 3)
   error in <...>/init-mismatch.zeek, line 38: illegal table constructor element (1.2.3.4)
  +error in <...>/init-mismatch.zeek, line 7: undeclared variable (b)
  =======================================
```

Reply:
I switched it to use `init->Error()` but removed the return.